File: frontend/Assets/Scripts/DataTransfer/Server/LobbyStatusCmd.cs
using System;
using System.IO;
using Attributes;
using Utils;


namespace DataTransfer.Server {

    [DTO (DTOCode.LobbyStatus)]
    public class LobbyStatusCmd : ServerCommand {

        public int HubId   { get; private set; }
        public int Players { get; private set; }

        public static event Action <LobbyStatusCmd> OnReceived;


        public override void ReadMembers (BinaryReader reader) {
            HubId   = reader.ReadByte ();
            Players = reader.ReadByte ();
        }


        public override void Execute () {
            OnReceived._ (this);
        }

    }

}

File: backend/src/main/java/dto/server/LobbyStatusCmd.java
package dto.server;

import dto.DTOCode;
import dto.DTOs;
import io.netty.buffer.ByteBuf;


@DTOCode(DTOs.LOBBY_STATUS)
public class LobbyStatusCmd extends ServerCommand {

    private final int lobbySize;
    private final int playersCount;


    public LobbyStatusCmd (int lobbySize, int playersCount) {
        this.lobbySize = lobbySize;
        this.playersCount = playersCount;
    }


    @Override
    public void writeMembers (ByteBuf byteBuf) {
        byteBuf
            .writeByte(lobbySize)
            .writeByte(playersCount);
    }
}
